packages/shorebird_cli/lib/src/commands/preview_command.dart

```diff
@@ -0,0 +1,186 @@
+        'app-id',
+        help: 'The ID of the app to preview the release for.',
+      )
+      ..addOption(
+        'release-version',
+        help: 'The version of the release (e.g. "1.0.0").',
+      );
+  }
+
+  @override
+  String get name => 'preview';
+
+  @override
+  String get description => 'Preview a specific release on a device.';
+
+  @override
+  Future<int> run() async {
+    try {
+      await validatePreconditions(checkUserIsAuthenticated: true);
+    } on PreconditionFailedException catch (error) {
+      return error.exitCode.code;
+    }
+
+    Future<String?> promptForApp() async {
+      final apps = await codePushClientWrapper.getApps();
+      if (apps.isEmpty) return null;
+      final app = logger.chooseOne(
+        'Which app would you like to preview?',
+        choices: apps,
+        display: (app) => app.displayName,
+      );
+      return app.appId;
+    }
+
+    Future<String?> promptForReleaseVersion(String appId) async {
+      final releases = await codePushClientWrapper.getReleases(appId: appId);
+      if (releases.isEmpty) return null;
+      final release = logger.chooseOne(
+        'Which release would you like to preview?',
+        choices: releases,
+        display: (release) => release.version,
+      );
+      return release.version;
+    }
+
+    const platform = 'android';
+    final appId = results['app-id'] as String? ?? await promptForApp();
+
+    if (appId == null) {
+      logger.info('No apps found');
+      return ExitCode.success.code;
+    }
+
+    final releaseVersion = results['release-version'] as String? ??
+        await promptForReleaseVersion(appId);
+
+    if (releaseVersion == null) {
+      logger.info('No releases found');
+      return ExitCode.success.code;
+    }
+
+    final previewDirectory = cache.getPreviewDirectory(appId);
+    final aabPath = p.join(
+      previewDirectory.path,
+      '${platform}_$releaseVersion.aab',
+    );
+    final apksPath = p.join(
+      previewDirectory.path,
+      '${platform}_$releaseVersion.apks',
+    );
+
+    if (!File(aabPath).existsSync()) {
+      final downloadArtifactProgress = logger.progress('Downloading release');
+      try {
+        final release = await codePushClientWrapper.getRelease(
+          appId: appId,
+          releaseVersion: releaseVersion,
+        );
+        final releaseAabArtifact =
+            await codePushClientWrapper.getReleaseArtifact(
+          releaseId: release.id,
+          // TODO(felangel): add iOS support
+          arch: 'aab',
+          platform: platform,
+        );
+
+        await releaseAabArtifact.url.download(aabPath);
+        downloadArtifactProgress.complete();
+      } catch (error) {
+        downloadArtifactProgress.fail('$error');
+        return ExitCode.software.code;
+      }
+    }
+
+    final extractMetadataProgress = logger.progress('Extracting metadata');
+    late String package;
+    try {
+      package = await bundletool.getPackageName(aabPath);
```

Comment:
Are we downloading the aab for any reason other than to get the package name?

Reply:
we need it to generate the apks which we install/run on the device.